src/control/server/ctl_smd_rpc.go

```diff
@@ -334,6 +343,33 @@ func addManageRespIDOnFail(log logging.Logger, res *ctlpb.SmdManageResp_Result,
 	}
 }
 
+// Retry dev-replace requests as state propagation may take some time after set-faulty call has
+// been made to manually trigger a faulty device state.
+func replaceDevRetryBusy(ctx context.Context, log logging.Logger, e Engine, req proto.Message) (res *ctlpb.SmdManageResp_Result, err error) {
+	for try := uint(0); try < uint(maxDevReplaceRetries); try++ {
+		res, err = sendManageReq(ctx, &e, drpc.MethodReplaceStorage, req)
+		if err != nil {
+			return
+		}
+		if daos.Status(res.Status) != daos.Busy {
+			break
+		}
+
+		backoff := common.ExpBackoff(baseDevReplaceBackoff, uint64(try),
```

Comment:
From what I understand, we will have an extra useless sleep when we consume all the `try` counter.

Reply:
If you mean there are 2 iterations with the same duration between then yes I also noticed this and have asked @mjmac previously in slack if this is intentional behaviour. Any relevant change is out of the scope of this PR however.

Reply:
I think the concern is that after the last attempt (try == maxDevReplaceRetries - 1), we will sit and wait even though we've exhausted all retries. Not a case I'd necessarily expect to see a lot of, but it could be confusing to debug.

Reply:
fix for this applied on https://github.com/daos-stack/daos/pull/12944